src/platform/screen.c

```diff
@@ -1,22 +1,78 @@
 #include "platform/screen.h"
 
-#include "SDL.h"
-
 #include "game/settings.h"
 #include "graphics/graphics.h"
 #include "graphics/screen.h"
 
+#include "SDL.h"
+
 static struct {
     SDL_Window *window;
     SDL_Renderer *renderer;
     SDL_Texture *texture;
-} SDL = {0, 0, 0};
+} SDL;
 
 static struct {
     int x;
     int y;
+    int width;
+    int height;
 } window_pos;
 
+
+static struct {
+    const int WIDTH;
+    const int HEIGHT;
+} MINIMUM = { 640, 480 };
+
+static int get_title_height()
+{
+#if SDL_COMPILEDVERSION < SDL_VERSIONNUM(2,0,5)
+    return setting_fullscreen() ? 0: 35;
+#else
+    int top;
+    SDL_GetWindowBordersSize(SDL.window, &top, NULL, NULL, NULL);
+    return top;
+#endif
+}
+
+static void position_window(int width, int height)
+{
+    int title_height = get_title_height();
+
+    if ((!window_pos.x && !window_pos.y) || !window_pos.width || !window_pos.height) {
+        platform_screen_center_window();
+    } else if (width != window_pos.width || height != window_pos.height) {
+        window_pos.x = window_pos.x + (window_pos.width / 2) - (width / 2);
+        window_pos.y = window_pos.y + (window_pos.height / 2) - (height / 2);
+    }
+
+    SDL_Rect monitor_coordinates;
```

Comment:
Why do we need this complex calculation?

Reply:
I'm guessing you're talking about the calculation with the `monitor_coordinates`. This calculation is to make sure the new window position is within the bounds of all existing monitors.

Reply:
Do we really want to do that ourselves? Why not delegate it to the window manager of the operating system? On macOS you have to consider the dock and global menu bar, on Windows the task bar can be on every edge, and Linux is even more flexible...

Reply:
The window manager unfortunately does nothing to prevent a window from being positioned outside the monitor range.

Regarding the task bar or the global menu bar, SDL has a function called [`SDL_GetDisplayUsableBounds`](https://wiki.libsdl.org/SDL_GetDisplayUsableBounds) that gets the valid bounds taking those into account, but it's only compatible with SDL 2.0.5.

Reply:
I've played around with this on my mac and sometimes the window moves in unexpected ways when hitting F7/8/9 repeatedly. I haven't been able to find a pattern.
I've now fixed the positioning issue with just a simple `SDL_GetWindowPos` before resizing the window. This keeps the top left corner of the window in the same place. While I think that moving the window so the contents stay in the same place is nice, I don't think the complexity of the code (and weird bugs it may cause on different platforms) justifies it.